_scripts/generate-release-notes

```
@@ -0,0 +1,458 @@
+files_images: list[str] = []
+
+
+class NoteImageParser(HTMLParser):
+    images: list[tuple[str, tuple[int, int]]] = []
+
+    def handle_starttag(self, tag, attrs):
+        if tag != "img":
+            return
+
+        for attr, val in attrs:
+            if attr == "src" and val is not None:
+                self.images.append((val, self.getpos()))
+
+
+def xdg_home(
+    subdir: str, envvar: str, *components: str, override: str | None = None
+) -> str:
+    path = override and os.getenv(override)
+
+    if not path:
+        directory = os.getenv(envvar)
+        if not directory:
+            directory = os.path.join(os.path.expanduser("~"), subdir)
+        path = os.path.join(directory, *components)
+
+    return path
+
+
+def xdg_config_home(*components: str, envvar: str | None = None) -> str:
+    return xdg_home(".config", "XDG_CONFIG_HOME", *components, override=envvar)
+
+
+def slugify(title: str) -> str:
+    return re.sub(r"[^0-9a-z]", "-", title.lower())
+
+
+def markdown_filename() -> str:
+    date_str = datetime.datetime.now().strftime("%Y-%m-%d")
+    return f"{date_str}-cockpit-{cockpit_version}.md"
+
+
+def oxfordize(parts: list[str]) -> str:
+    if len(parts) <= 1:
+        return "".join(parts)
+    if len(parts) == 2:
+        return " and ".join(parts)
+    return f"{', '.join(parts[:-1])}, and {parts[-1]}"
+
+
+def get_json(url: str) -> Any:
+    headers = {"user-Agent": USER_AGENT}
+    with contextlib.suppress(FileNotFoundError):
+        token = Path(xdg_config_home("cockpit-dev/github-token")).read_text()
+        if token is not None:
+            headers["Authorization"] = f"token {token.strip()}"
+
+    try:
+        # Specify agent as some websites otherwise block requests
+        req = Request(url=url, headers=headers)
+        resp = urlopen(req)
+        return json.loads(resp.read())
+    except urllib.error.URLError as e:
+        print(f"Could not fetch data from the GitHub API ({e})")
+        return None
```

AIs tend to do that. But I strongly suggest to drop these kind of exception handlers. We'd rather have the script crash and fix it than silently ignoring errors and creating incomplete output.